[INJIMOB-3409] add: support for did web, jwk, key Public key resolution

KiruthikaJeyashankar opened the pull request:
Supported did methods
- web
- jwk
- key

did web verification materials supported
- jwk
- multibase
- pem
- hex

Supported key types
- Ed25519

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: return PublicKeyType in publickeyResolver

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: fix usage of publicKeyType

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] add: did-web publicKeyJwk support

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] add: did jwk support

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] add: did-web support with public key hex and pem

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: update supported verification materials for did web

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: update JWK class

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] fix: did key resolving

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] test: add tests for did resolvers

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] test: enable unsupported verification material test

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: extract constants to file

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: structure folder for did related resolvers

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: modify condition to use enum

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: structure folder for publickKeyTy[pe

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: public key resolver types

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: separet classes to files

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: remove unused / duplicate code

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: modify error messages

Other refactorings
- remove redundant base64url to base64 conversions by extracting it to extension of string in a file

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: remove duplicate funcitons

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: modify error messages in Did key resolver

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: sync test files folder struct with src files

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: extract duplicate to funciton

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] test: add test for key resolver utils

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] test: improve assertion

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: rm unused function

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: rm debug logs

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: move tests to related files

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] test: add tests for JWSHandler

Other changes
- structure tests

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3390] refactor: remove duplicate function

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3390] refactor: rename test extentions with prefix XCT

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3390] refactor: extract assertion for testing ed key

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: restucture public key resolver class design

Key changes
- One publicKeyResolver can resolve more than one inputs

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: modify JWSHandler to accept jws as verify fn param

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

swatigoel commented on Sources/OpenID4VP/AuthorizationRequest/ClientMetadata/Jwks.swift:
```diff
@@ -1,4 +1,6 @@
 import Foundation
+//import JSONWebKey
+//TODO: use JWK from jose-swift
```
what is these ToDos about?

KiruthikaJeyashankar replied:
As of now, JWK data class is maintained by the ovp lib itself, Was thinking if we can leverage on already available libraries for this. So added as a TODO at that moment

KiruthikaJeyashankar replied:
Will be taken as part of next story

swatigoel commented on Sources/OpenID4VP/Constants/PublicKeyVerificationMaterial.swift:
```diff
@@ -0,0 +1,15 @@
+//
+//  File.swift
+//  OpenID4VP
+//
+//  Created by Kiruthika J on 05/08/25.
+//
+
```
remove these comments

KiruthikaJeyashankar replied:
Done

swatigoel commented on Sources/OpenID4VP/NetworkManager/NetworkManager.swift:
```diff
@@ -48,10 +48,12 @@ public struct NetworkManager: NetworkManaging {
                     if let urlError = error.asAFError?.underlyingError as? URLError, urlError.code == .timedOut {
                         let exception = NetworkRequestException.networkRequestTimeout
                         OpenID4VPException.error(NetworkManager.logTag, exception)
+                        print("url failure: \(url)")
```
do we need this log?

KiruthikaJeyashankar replied:
Removed

swatigoel commented on Sources/OpenID4VP/NetworkManager/NetworkManager.swift:
```diff
@@ -48,10 +48,12 @@ public struct NetworkManager: NetworkManaging {
                     } else {
                         let exception = NetworkRequestException.networkRequestFailed(message: "\(error.localizedDescription)")
                         OpenID4VPException.error(NetworkManager.logTag, exception)
+                        print("url failure: \(url)")
```
do we need this log?

KiruthikaJeyashankar replied:
Removed

swatigoel commented on Sources/OpenID4VP/jwt/JWS/JWSHandler.swift:
```diff
@@ -5,43 +5,28 @@ import Base58Swift
+                    throw InvalidSignature(
+                        className: JWSHandler.className
+                    )
+                }
```
don't we need default case?

KiruthikaJeyashankar replied:
Not required @swatigoel as all publicKey types as per enum are handled
If default case is added as well, it will not be executed

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: remove debug lgs

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] add: workflow build to use iOS simulator

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] debug: list available workflow runners

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] fix: runner unavailable issue on push trigger

Cause:
- macos runner version update
Resolution:
- Pin Xcode app version to 15.4
Reference
- actions/runner-images#12520

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: print env details in worklflow

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>

KiruthikaJeyashankar committed:
[INJIMOB-3409] refactor: replace PublicKeyExtractionFailed -> PublicKeyResolutionFailed

Signed-off-by: KiruthikaJeyashankar <kiruthikavjshankar@gmail.com>
